include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -525,7 +525,12 @@ REMARK(replay_output, none, "replay output file '%0': key '%1'", (StringRef, Str
 REMARK(output_cache_miss, none, "cache miss for input file '%0': key '%1'", (StringRef, StringRef))
 
 // CAS related diagnostics
-ERROR(error_cas, none, "CAS error encountered: %0", (StringRef))
+ERROR(error_invalid_cas_id, none, "cannot parse CASID '%0': %1", (StringRef, StringRef))
+ERROR(error_cas, none, "CAS error encountered during %0: %1", (StringRef, StringRef))
+ERROR(error_cas_fs_creation, none, "cannot create CAS file system: %0", (StringRef))
+ERROR(error_cache_key_creation, none, "cannot create cache key for compilation %0: %1", (StringRef, StringRef))
+ERROR(error_cas_file_ref, none, "failed to get CAS ref for file: %0", (StringRef))
```

Comment:
I'm wondering if "cannot find file %0 in cas filesystem" or something similar might be better, but I'm not completely clear how this error can happen.